internal/controller/common/action/transitions/to_pending_phase.go

```diff
@@ -0,0 +1,40 @@
+package transitions
+
+import (
+	"context"
+
+	"github.com/securesign/operator/internal/apis"
+	"github.com/securesign/operator/internal/controller/common/action"
+	"github.com/securesign/operator/internal/controller/constants"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewToPendingPhaseAction[T apis.ConditionsAwareObject](componentSupplier func(T) []string) action.Action[T] {
```

Comment:
You can define a new type ComponentSupplier. Like:
```go
type ComponentSupplier [T apis.ConditionsAwareObject]func(T) []string 
```

```suggestion
func NewToPendingPhaseAction[T apis.ConditionsAwareObject](supplier ComponentSupplier[T]) action.Action[T] {
```